File: src/my_project/easy_problems/from101to150/check_arithmetic_progression.py
from typing import List 

class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:

        arr.sort()

        solution: bool = True

        if len(arr) == 1:
            return solution

        temp = arr[1] - arr[0]
        for i in range(len(arr)-1):
            if arr[i+1] - arr[i]  != temp:
                solution = False 
                return solution

        return solution 


solution = Solution()

print(solution.canMakeArithmeticProgression(arr = [1,2,4]))

File: my_project_typescript/easy_problems_typescript/from1to50/check_arithmetic_progression.ts
function canMakeArithmeticProgression(arr: number[]): boolean {

    arr = arr.sort((x,y) => x <= y ? -1: 1)

    let solution: boolean = true;

    if (arr.length === 1){
        return solution;
    }

    let temp = arr[1] - arr[0]

    for (let i = 0; i < arr.length -1; i++){
        if (arr[i+1] - arr[i] !== temp){
            solution = false;
            return solution;
        }
    }

    return solution;

};

console.log(canMakeArithmeticProgression([3,5,1]))